ResearchKit/Common/ORKImageCaptureView.m

```diff
@@ -0,0 +1,257 @@
+const CGFloat CONTINUE_ALPHA_TRANSLUCENT = 0.5;
+const CGFloat CONTINUE_ALPHA_OPAQUE = 0;
+
+- (void)updateConstraints {
+    // Remove the existing constraints
+    if (self.mconstraints) {
+        [self removeConstraints:self.mconstraints];
+        [self.mconstraints removeAllObjects];
+    }
+    
+    NSDictionary *dictionary = NSDictionaryOfVariableBindings(_previewView, _continueSkipContainer, _headerView);
+    ORKEnableAutoLayoutForViews([dictionary allValues]);
+    if (_error) {
+        // If we have an error to show, do not display the previewView at all
+        [self.mconstraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[_headerView]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+        [self.mconstraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[_continueSkipContainer]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+        [self.mconstraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_headerView]-[_continueSkipContainer]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+    } else {
+        // If we do not have an error to show, layout the previewView and continueSkipContainer
+        [self.mconstraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[_previewView]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+        [self.mconstraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[_continueSkipContainer]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+    
+        // Float the contine view over the previewView if in landscape to give more room for the preview
+        if ([self floatContinueView]) {
+            [self.mconstraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_previewView]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+            [self.mconstraints addObject:[NSLayoutConstraint constraintWithItem:_continueSkipContainer attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1 constant:0]];
+            _continueSkipContainer.backgroundColor = [_continueSkipContainer.backgroundColor colorWithAlphaComponent:CONTINUE_ALPHA_TRANSLUCENT];
+        } else {
+            [self.mconstraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_previewView]-[_continueSkipContainer]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+            _continueSkipContainer.backgroundColor = [_continueSkipContainer.backgroundColor colorWithAlphaComponent:CONTINUE_ALPHA_OPAQUE];
+        }
+    }
+    
+    [self addConstraints:self.mconstraints];
+    
+    [super updateConstraints];
+}
+
+- (BOOL)floatContinueView {
```

Comment:
Can this be renamed to isApplicationInLandscapeOrientation or something? Or have it inline, since it’s just one line?